lib/index.js

```diff
@@ -414,9 +414,12 @@ var Client = module.exports = function(config) {
     };
 
     function getPage(link, which, headers, callback) {
+        var self = this;
         var url = getPageLinks(link)[which];
-        if (!url)
-            return callback(new error.NotFound("No " + which + " page found"));
+        if (!url) {
+            var urlErr = new error.NotFound("No " + which + " page found");
+            return self.Promise && !callback ? self.Promise.reject(urlErr) : callback(urlErr);
```

Comment:
I'll leave it be 😄 , I'm sure my lazy developer habits will come back to haunt me.